pkg/asset/cluster/tfvars.go

```diff
@@ -325,6 +325,24 @@ func (t *TerraformVariables) Generate(parents asset.Parents) error {
 		}
 
 		preexistingnetwork := installConfig.Config.Azure.VirtualNetwork != ""
+
+		var bootstrapIgnStub, storageAccountName string
+		if installConfig.Azure.CloudName == azure.StackCloud {
+			// Our Azure Terraform config uses a random string to name storage accounts as the names must be globally unique & only alphanumeric.
+			// In order to create the bootstrap stub in advance, that name needs to be deterministic, so we can use the
+			// infrastructure ID suffix, which is also a random string. Keep the "cluster" prefix to be consistent with public Azure.
+			infraIDSuffix := strings.Split(clusterID.InfraID, "-")[1]
+			storageAccountName = fmt.Sprintf("cluster%s", infraIDSuffix)
+
+			storageSuffix := strings.TrimPrefix(installConfig.Azure.ARMEndpoint, "https://management.")
```

Comment:
Should probably change this based on this comment: https://github.com/openshift/installer/pull/5103/files#r725223700

I might need to update the metadata.